subsecs.h

```diff
@@ -0,0 +1,282 @@
+  { // with a width of zero, nothing happens;
+    return os;
+  }
+};
+
+
+template <class SrcPeriod>
+struct time_subsec_put {
+  // A more relaxed condition than the (num < den)
+  // Any already reduced rational fraction with den>1 will
+  // cause decimals (if it didn't, then it would be an integer)
+  static CONSTDATA bool admits_subsecs= (SrcPeriod::den!=1);
+
+  // refuse to deal with values greater than seconds, thus require
+  // SrcPeriod::num strictly less than SrcPeriod::den.
+  // Heuristic: denominator also carry the required precision
+  static CONSTDATA std::uintmax_t max_val=
+      POW10(CEIL_LOG10(SrcPeriod::den))
+  ;
+
+  // the 'width' occupied by the digits.
+  static CONSTDATA unsigned precision=CEIL_LOG10(max_val);
+
+  // This is the space needed to output the subseconds value in full precision,
+  // including the decimal dot.
+  static CONSTDATA unsigned space_req=precision+1;
+
+
+  // precomputed scale factor to be applied to received counts
+  static CONSTDATA long double scale_factor=
+    1.0L*max_val*SrcPeriod::num/SrcPeriod::den
+  ;
+
+  // Because we are supposed to deal with **sub**-seconds, anything that goes
+  // above max_val is discarded (we only keep the lesser-significant digits)
+  template <typename Rep> Rep condition_value(Rep count) {
+    std::uintmax_t scaled=static_cast<std::uintmax_t>(scale_factor*count);
+    return static_cast<Rep>(scaled % max_val);
+  }
+
+  // for periods that do have subseconds
+  template <typename Rep,typename CharT>
+  void insert(
+    std::basic_string<CharT>& dest,
+    std::size_t& inspos,
+    std::chrono::duration<Rep, SrcPeriod> duration,
+    const std::locale& loc
+  )
+  {
+    if(admits_subsecs) {
+      // scale the count
+      Rep count=condition_value(duration.count());
+      // make room for the content in the destination
+      dest.insert(inspos, space_req, ' ');
+      // insert the decimal dot
+      CharT dotChar=std::use_facet<std::numpunct<CharT>>(loc).decimal_point();
+      dest[inspos]=dotChar;
+      inspos++;
+      // prepare the filler
+      decimals_fmt<precision, Rep> filler;
+      // output the value
+      filler.put(dest, inspos, count);
+    }
+  }
+
+  // In ostream
+  // Periods with subseconds
+  template <typename Rep, typename CharT=char>
+  std::basic_ostream<CharT>& put(
+      std::basic_ostream<CharT>& os,
+      std::chrono::duration<Rep,SrcPeriod> count
+  )
+  {
+    if(admits_subsecs) {
+      // scale the count
+      count=condition_value(count);
+      // insert the decimal dot
+      os << std::use_facet<std::numpunct<CharT>>(os.getloc()).decimal_point();
+      // prepare the filler
+      decimals_fmt<precision, Rep> filler;
+      // echo the value
+      filler.put(os, count);
+    }
+    return os;
+  }
+};
+#undef POW10
+#undef CEIL_LOG10
+
+} // namespace detail
+} // namespace date
+
+
+#undef CONSTDATA
+#undef CONSTCD11
+#undef CONSTCD14
+#undef NOEXCEPT
+
```

Comment:
I like that these work in C++11.  There's some solid work in here.  I don't like the separate header.  I'd like to fold this into "date.h".

Technically your copyright on this file is too strict (not open source).  I'll need your permission to put this code (or derivatives of it) into date.h which has the MIT open source license (and add your name to the copyright holders in that file).